mongo_connector/oplog_manager.py

```diff
@@ -183,12 +196,42 @@ def oplog_ns_set(self):
     def update_oplog_ns_set(self):
         self._oplog_ns_set = []
         if self.namespace_set:
-            self._oplog_ns_set.extend(self.namespace_set)
+            for ns in self.namespace_set:
+                if "*" in ns:
+                    # if namespace_set contains *, change to regexp
+                    reg_ns = re.sub(r"(\.)","\.", ns)
+                    reg_ns = re.sub(r"(\*)", ".*", reg_ns)
+                    LOG.debug("OplogThread: Replace namespace %s to %s"  %(ns, reg_ns))
+                    reg = re.compile(reg_ns)
+                    self._oplog_ns_set.append(reg)
+                else:
+                    self._oplog_ns_set.append(ns)
+                    self._oplog_ns_set.append(ns.split('.', 1)[0] + '.$cmd')
+
             self._oplog_ns_set.extend(self.gridfs_files_set)
-            self._oplog_ns_set.extend(set(
-                ns.split('.', 1)[0] + '.$cmd' for ns in self.namespace_set))
             self._oplog_ns_set.append("admin.$cmd")
 
+    @property
+    def oplog_ex_ns_set(self):
+        try:
+            return self._oplog_ex_ns_set
+        except AttributeError:
```

Comment:
This shouldn't ever happen, since we call `self.oplog_ex_ns_set = ...` in the constructor, which sets `self._oplog_ex_ns_set`.